### PR TITLE
feat: expose more properties of Kafka topcis in `kafka-topic` component module

### PR DESCRIPTION
**Pull request by JoseFMP**
Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

After importing some of the topics from that were mirrored from Strimzi into Confluent cloud, I found out there is some drift in these topics. In this PR I expose in the 'topic' component module those properties so that we can set them from outside and therefore avoid drift, and actually set values that might make sense.

- - -
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-confluent-kafka/21)

### CODE REVIEW

**Review by poom — APPROVED** [review]

**Review by xoxwaw — APPROVED** [review]

File: modules/kafka-topic/inputs.tf [hunk 1]
```diff
@@ -31,12 +31,45 @@ variable "delete_policy" {
...
+variable "delete_retention_ms" {
+  type        = string
+  description = "The amount of time to retain delete tombstone markers for log compacted topics."
+  default     = null
+}
+
+variable "max_compaction_lag_ms" {
+  type        = string
+  default     = null
+  description = "The maximum time a message will remain ineligible for compaction in the log. Only applicable for logs that are being compacted."
```
Comment (MXfive) [inline modules/kafka-topic/inputs.tf]: Can we sort the attributes consistently please.

File: examples/create-env-cluster-topics/main.tf [hunk 2]
```diff
@@ -90,7 +90,14 @@ module "honest_labs_kafka_topic_example_1" {
   service_account_id           = module.kafka_topic_service_account.service_account_id
   topic_name                   = "squad_raw_service_example_1_entity"
   connector_service_account_id = module.honest_labs_connector_service_account.service_account_id
-  depends_on                   = [module.cluster_admin_privilege_service_account]
+
+  max_compaction_lag_ms              = "9223372036854775807"
```
Comment (MXfive) [inline examples/create-env-cluster-topics/main.tf]: Is this just a random number, or a real duration?  Let's use a real number as the example, with a comment showing the duration in more readable units.

File: examples/create-env-cluster-topics/main.tf [hunk 3]
```diff
@@ -90,7 +90,14 @@ module "honest_labs_kafka_topic_example_1" {
...
+  delete_retention_ms                = "100"
+  message_timestamp_difference_mx_ms = "9223372036854775807"
+  min_compaction_lag_ms              = "0"
+  retention_bytes = "-1"
```
Comment (MXfive) [inline examples/create-env-cluster-topics/main.tf]: Indentation.

**Review by MXfive — CHANGES_REQUESTED** [review]

File: examples/create-env-cluster-topics/main.tf (hunk 2 above)
Comment (JoseFMP) [reply examples/create-env-cluster-topics/main.tf]: It is a real number. That's why I used in the test

File: examples/create-env-cluster-topics/main.tf (hunk 3 above)
Comment (JoseFMP) [reply examples/create-env-cluster-topics/main.tf]: Corrected now.

**Review by MXfive — APPROVED** [review]

**Pull request merged**